components/diagnose/scan.go

```diff
@@ -213,6 +214,19 @@ func Scan(ctx context.Context, opts ...OpOption) error {
 						}
 					}
 				}
+
+				if op.checkIb {
+					threshold := infiniband.SupportsInfinibandPortRate(output.GPUProductName())
```

Comment:
We’d like to wrap this code into a function, and later we should refactor the Scan function to improve its readability and maintainability.

Reply:
Yes, def. Let's keep this for now, since we need a separate parameter to pass the auto detected thresholds